src/events/interactionCreate.ts

```diff
@@ -0,0 +1,19 @@
+import { event, Events } from '../utils/index.js'
+import commands from '../commands/index.js'
+
+/**
+ * Interaction creation listener for the client
+ * @param interaction the interaction received from the server
+ */
+export default event(Events.InteractionCreate, async ({ log, client }, interaction) => {
+    if (!interaction.isCommand() || !interaction.isChatInputCommand()) return
+    
+    log(`Interaction called \'${interaction.commandName}\' from ${interaction.client.user.tag}.`)
+
+    // ensure command exists
+    const command = commands.find(command => command.name === interaction.commandName)
+    if (!command) return
+
+    // take all commands and run the command
+    command.run(client, interaction)
```

Comment:
How is this line running 'all' commands? From my understanding, this runs a single command found from the collection of 'commands' imported at the top.

Reply:
Poor comment choice, I probably had something there before that did that before I committed to the branch, I can change it.